common/src/warp_runner/manager/commands/raygun_commands.rs

```diff
@@ -106,8 +116,29 @@ pub async fn handle_raygun_cmd(
             };
             let _ = rsp.send(r);
         }
-        RayGunCmd::SendMessage { conv_id, msg, rsp } => {
-            let r = messaging.send(conv_id, None, msg).await;
+        RayGunCmd::SendMessage {
+            conv_id,
+            msg,
+            attachments,
+            rsp,
+        } => {
+            let r = if attachments.is_empty() {
+                messaging.send(conv_id, None, msg).await
+            } else {
+                messaging.attach(conv_id, attachments, msg).await
+            };
+
+            let _ = rsp.send(r);
+        }
+        RayGunCmd::DownloadAttachment {
+            conv_id,
+            msg_id,
+            file_name,
+            directory,
+            rsp,
+        } => {
+            let pb = directory.join(&file_name);
```

Comment:
It might be better for `DownloadAttachment` to use a single `PathBuf` representing the location and file where it would be stored at. I do see the reason for it and its not blocking to the PR, but just generally a suggestion.

Reply:
Agreed. Especially if the user wants to rename the file when downloading it.